lib/ocpp/v201/smart_charging.cpp

```diff
@@ -77,36 +77,58 @@ ProfileValidationResultEnum SmartChargingHandler::validate_tx_profile(const Char
  * - K01.FR.20
  * - K01.FR.34
  * - K01.FR.43
- * - K01.FR.45
  * - K01.FR.48
  */
-ProfileValidationResultEnum SmartChargingHandler::validate_profile_schedules(const ChargingProfile& profile) const {
-    auto schedules = profile.chargingSchedule;
+ProfileValidationResultEnum
+SmartChargingHandler::validate_profile_schedules(ChargingProfile& profile,
```

Comment:
At first I was thinking this could have `const` arguments but it also fixes some stuff like the default to 3 phases.

I would argue we should either change the name of the method to reflect that or split the validation and applying assumptions in different methods.

Reply:
If we change the name of the method, what would you suggest?

Reply:
I actually don't know without it becoming a mess of a method name. I am generally not in favor of having validations and getter functions modify stuff but we do it more in this library so lets leave it be.

As a compromise you could maybe update the doxygen for the method to reflect that it is actually setting stuff to default if left empty.